backend/api/communication/src/main/java/co/airy/core/api/communication/ConversationsController.java

```diff
@@ -168,6 +160,83 @@
         return ResponseEntity.noContent().build();
     }
 
+    @PostMapping("/conversations.addNote")
+    ResponseEntity<?> conversationAddNote(@RequestBody @Valid ConversationAddNoteRequestPayload payload) {
+        final String conversationId = payload.getConversationId().toString();
+        final String text = payload.getText();
+        if (text != null && (text.length() == 0 || text.length() >50000)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new RequestErrorResponsePayload("Text length must be >0 and <50000"));
+        }
+
+        final ReadOnlyKeyValueStore<String, Conversation> store = stores.getConversationsStore();
+        final Conversation conversation = store.get(conversationId);
+
+        if (conversation == null) {
+            return ResponseEntity.notFound().build();
+        }
+
+        final Metadata metadata = newConversationNote(conversationId, text);
+
+        try {
+            stores.storeMetadata(metadata);
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(new RequestErrorResponsePayload(e.getMessage()));
+        }
+
+        return ResponseEntity.noContent().build();
+    }
+
+    @PostMapping("/conversations.deleteNote")
```

Comment:
Per user request (#2353 ) we now use hyphenation for multi word path parts. So these routes should be `/conversations.add-note` and `/conversations.delete-note`